docs/introduction/Sidechain-SDK-extension.rst

```diff
@@ -83,34 +87,33 @@ The most relevant methods of this class are detailed below:
 
   The two methods define the id of the closed box to be opened and the proof that unlocks the proposition for that box. When a box is unlocked and opened, it is spent or "burnt", i.e. it stops existing; as such, it will be removed from the wallet and the blockchain state. As a reminder, a value inside a box cannot be "updated": the the process requires to spend the box and create a new one with the updated values.
 
-- ``public List<NoncedBox<Proposition>> newBoxes()``
+- ``public List<Box<Proposition>> newBoxes()``
 
   This function returns the list of new boxes which will be created by the current transaction. 
   As a good practice, you should use the ``Collections.unmodifiableList()`` method to wrap the returned list into a not updatable Collection:
 
   ::
 
     @Override
-    public List<NoncedBox<Proposition>> newBoxes() {
-      List<NoncedBox<Proposition>> newBoxes =  .....  //new boxes are created here  
+    public List<Box<Proposition>> newBoxes() {
+      List<Box<Proposition>> newBoxes =  .....  //new boxes are created here
       //....
       return Collections.unmodifiableList(newBoxes);
     }   
 
 - ``public long fee()``
   Returns the fee to be paid to execute this transaction.
 
-- ``public long timestamp()``
-  Returns the timestamp of the transaction creation.
-  As a good practice, timestamp should be created outside transaction, passed in the transaction's constructor, and returned here.
-
 - ``public byte transactionTypeId()``
   Returns the type of this transaction. Each custom transaction must have its own unique type.
 
 - ``public boolean transactionSemanticValidity()``
   Confirms if a transaction is semantically valid, e.g. check that fee > 0, timestamp > 0, etc.
   This function is not aware of the state of the sidechain, so it can't check, for instance, if the input is a valid Box.
 
+SidechainNoncedTransaction has already implementation of newBoxes function. But it requires an implementation of abstract function getOutputData that provides list of output data of the transaction.
+AbstractRegularTransaction requires an implementation of getCustomOutputData for retrieving output custom data of the transaction. Output of other data in AbstractRegularTransaction is already collected in getOutputData function, which also uses getCustomOutputData.
```

Comment:
AbstractRegularTransaction requires the implementation of getCustomOutputData for retrieving output custom data of the transaction. The output of other data in AbstractRegularTransaction is already collected in the getOutputData function, which also uses getCustomOutputData.